modules/drivers/snowflake/test/metabase/driver/snowflake_test.clj

```diff
@@ -298,3 +314,36 @@
     ;; Special characters
     (is (= "USE ROLE \"Role.123\";"   (driver.sql/set-role-statement :snowflake "Role.123")))
     (is (= "USE ROLE \"$role\";"      (driver.sql/set-role-statement :snowflake "$role")))))
+
+(deftest remark-test
+  (testing "Queries should have a remark appended to them with additional metadata"
+    (mt/test-driver :snowflake
+      (let [expected (for [line ["SELECT"
+                                 "  \"PUBLIC\".\"users\".\"id\" AS \"id\","
+                                 "  \"PUBLIC\".\"users\".\"name\" AS \"name\","
+                                 "  \"PUBLIC\".\"users\".\"last_login\" AS \"last_login\""
+                                 "FROM"
+                                 "  \"{{database_prefix}}test-data\".\"PUBLIC\".\"users\""
+                                 "WHERE"
+                                 "  (\"PUBLIC\".\"users\".\"id\" = 1)"
+                                 "  OR (\"PUBLIC\".\"users\".\"id\" = 2)"
+                                 "  OR (\"PUBLIC\".\"users\".\"id\" = 3)"
+                                 "LIMIT"
+                                 "  2000 -- {\"pulseId\":null,\"serverId\":\"{{site-uuid}}\",\"client\":\"Metabase\",\"queryHash\":\"cb83d4f6eedc250edb0f2c16f8d9a21e5d42f322ccece1494c8ef3d634581fe2\",\"queryType\":\"query\",\"cardId\":1234,\"dashboardId\":5678,\"context\":\"ad-hoc\",\"userId\":1000,\"databaseId\":1}"]]
```

Comment:
The tests are failing because `databaseId` is not usually `1`, you should get it as `(mt/id)`.

Also, testing the JSON representation is brittle, because Clojure maps are generally not guaranteed to keep the order. Small maps are implemented as `array-map`s which do keep the order, but after a certain size (above about 10 entries, so we are close) hash-maps are used and those don't.

So either use `array-map` in `snowflake.clj`, or (probably more complicated but better) parse the returned JSON string and compare Clojure maps.

Reply:
Those are excellent points, thank you! I've pushed a new commit trying to address them (with the help from ChatGPT 😅).

By the way, what happens once we are done reviewing this PR? Will you merge this PR yourself? Or will you merge your mirror PR? I'm just asking so I know if I should do anything myself. In case I'm supposed to merge it myself, can I "squash & merge" since I have a lot of small commits?

Reply:
We squash merge your PR. The mirror PR is there just to run the tests.